sqlglot/parser.py

```diff
@@ -3612,6 +3622,8 @@ def _parse_types(
                         type_name = f"{type_name}.{self._advance_any() and self._prev.text}"
 
                     return exp.DataType.build(type_name, udt=True)
+                elif allow_functions:
+                    return identifier
```

Comment:
do you have a test case that hits this line?

Reply:
yep, all the cases with `AggregateFunction(func, ...)` `func` is a bare function name identifier

Reply:
Maybe it needs to be restricted to a very specific set of parent data types though...

Reply:
doesn't SimpleAggregateFunction(sum, Float64) get parsed by line 3602, wouldn't it turn SimpleAggregateFunction into a function?

Reply:
ah i see what you're saying, yea, we'd only need to parse functions when it's simpleaggfunc or the other

Reply:
`SimpleAggregateFunction` is `nested=True` data type.
We are now in un-nesting part at that point.

Reply:
Yeah. I think flag is ugly.
So you're suggesting to add a new data type family? And then parse it separately?
Putting only `SimpleAggregateFunction` and `AggregateFunction` there?

Reply:
something like:
```python
if self._match_set(self.AGGREGATE_TYPE_TOKENS):
    # do the bare identifier + function stuff

```

Reply:
Ok, I think I've got a better idea. Will propose tomorrow.

Reply:
exactly

Reply:
Hey @pkit, any updates on this PR?

Reply:
Soon. It was my birthday yesterday. Lol.

Reply:
happy birthday!

Reply:
Ahah gotcha - happy birthday! 😄

Reply:
Thanks!
pushed a new one